cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -328,6 +332,16 @@ type ExternalEtcd struct {
 	KeyFile string
 }
 
+// GetHTTPEndpoints returns the HTTP endpoints for external etcd communication.
+// If HTTPEndpoints is provided and not empty, it returns HTTPEndpoints.
+// Otherwise, it returns the regular Endpoints.
+func (e *ExternalEtcd) GetHTTPEndpoints() []string {
+	if len(e.HTTPEndpoints) > 0 {
+		return e.HTTPEndpoints
+	}
+	return e.Endpoints
+}
+
```

Comment:
given we apply defaulting on config load, can it be done here:
https://github.com/SataQiu/kubernetes/blob/c080adc2364aee7a275a9cccaf96ca8eef785c2b/cmd/kubeadm/app/apis/kubeadm/v1beta4/defaults.go#L124
instead of adding a new GetHTTPEndpoints() method?